### PR TITLE
Add auto ssl

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-server-mongodb-operator?pullRequest=184) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-server-mongodb-operator?pullRequest=184) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: delgod<br/>:x: nonemax<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-server-mongodb-operator?pullRequest=184) it.</sub>

File: deploy/rbac.yaml [hunk 1]
```diff
@@ -79,3 +79,36 @@ roleRef:
...
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: percona-server-mongodb-operator-global
+subjects:
+- kind: Group
+  name: system:serviceaccounts
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  name: ppercona-server-mongodb-operator
```
Comment (delgod) [inline deploy/rbac.yaml]:
```suggestion
  name: percona-server-mongodb-operator
```

File: pkg/controller/perconaservermongodb/psmdb_controller.go [hunk 2]
```diff
@@ -319,13 +395,13 @@ func (r *ReconcilePerconaServerMongoDB) reconcileStatefulSet(arbiter bool, cr *a
...
-	t := true
+	t := cr.Spec.UnsafeConf
 	sfsSpec.Template.Spec.Volumes = append(sfsSpec.Template.Spec.Volumes,
 		corev1.Volume{
 			Name: "ssl",
 			VolumeSource: corev1.VolumeSource{
 				Secret: &corev1.SecretVolumeSource{
-					SecretName: sfs.Name + "-ssl",
+					SecretName: cr.Spec.SSLSecretName,
 					Optional:   &t,
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/psmdb_controller.go]:
```suggestion
					Optional:   &cr.Spec.UnsafeConf,
```

File: pkg/controller/perconaservermongodb/psmdb_controller.go [hunk 3]
```diff
@@ -131,6 +132,13 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, fmt.Errorf("wrong psmdb options: %v", err)
 	}
 
+	if !cr.Spec.UnsafeConf {
+		err = r.reconsileSSL(cr, cr.Namespace)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf(`TLS secret handler return error: "%v". Please create your TLS secrets `+cr.Spec.SSLSecretName+` and `+cr.Spec.SSLInternalSecretName+` manually or setup cert-manager correctly`, err)
```
Comment (dAdAbird) [inline pkg/controller/perconaservermongodb/psmdb_controller.go]:
```suggestion
			return reconcile.Result{}, fmt.Errorf(`TLS secrets handler: "%v". Please create your TLS secrets `+cr.Spec.SSLSecretName+` and `+cr.Spec.SSLInternalSecretName+` manually or setup cert-manager correctly`, err)
```

**Pull request merged**